src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,197 @@
+using Distances
+using ComplexityMeasures: ProbabilitiesEstimator
+
+export StatisticalComplexity, minimum_complexity_entropy, maximum_complexity_entropy
+
+"""
+    StatisticalComplexity <: ComplexityEstimator
+    StatisticalComplexity([x]; kwargs...)
+
+An estimator for the statistical complexity and entropy according to [^Rosso2007], used with [`complexity`](@ref).
+
+## Keyword arguments
+- `est::ProbabilitiesEstimator = SymbolicPermutation()`: which estimator to use to get the probabilities
+- `dist<:SemiMetric = JSDivergence()`: the distance measure between the estimated probability
+    distribution and a uniform distribution with the same maximal number of bins
+
+## Description
+Statistical Complexity is defined as 
+```math
+C_q[P] = \\mathcal{H}_q\\cdot \\mathcal{Q}_q[P],
+```
+where `Q_q` is a "disequilibrium" obtained from a distance-measure and
+`H_q` a disorder measure.
+In the original paper[^Rosso2007], this complexity measure was defined
+via an ordinal pattern-based probability distribution, the Shannon entropy 
+and the Jensen-Shannon divergence as a distance measure.
+This implementation allows for a generalization of the
+complexity measure as developed in [^Rosso2013].
+Here, `H_q` can be the (q-order) Shannon-, Renyi or Tsallis
+entropy and Q_q based either on the Euclidean, Wooters, Kullback,
+q-Kullback, Jensen or q-Jensen distance.
+
+[^Rosso2007]: Rosso, O. A. et al. (2007). Distinguishing Noise from Chaos. 
+    Physical Review Letters 99, no. 15: 154102. https://doi.org/10.1103/PhysRevLett.99.154102.
+[^Rosso2013]: Rosso, O. A. (2013) Generalized Statistical Complexity: A New Tool for Dynamical Systems.
+
+"""
+Base.@kwdef struct StatisticalComplexity{E, D, H} <: ComplexityEstimator
+    dist::D = JSDivergence()
+    est::E = SymbolicPermutation()
+    entr::H = Renyi()
+
+    function StatisticalComplexity(dist::D, est::E, entr::H) where {E, D, H}
+        new{E, D, H}(dist, est, entr)
+    end
+    function StatisticalComplexity(x::AbstractVector;
+            dist::SemiMetric, est::ProbabilitiesEstimator,
+            entr::EntropyDefinition)
+        StatisticalComplexity(dist, est, entr)
+    end
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractVector{T}) where T
```

Comment:
Is it necessary to restrict to `AbstractVector` here? Some `ProbabilitiesEstimator`s also accept multivariate inputs (i.e. `AbstractStateSpaceSet`, which used to be `AbstractDataset`). Is the statistical complexity defined for multivariate data too?